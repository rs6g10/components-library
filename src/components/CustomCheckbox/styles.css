.checkbox {
    border: 1px solid black;
    height: 10px;
    width: 10px;
    display: inline-block;
}

.check {
    visibility: hidden;
    background: black;
    display: block;
    height: 100%;
}

.name {
    padding-left: 10px;
}
