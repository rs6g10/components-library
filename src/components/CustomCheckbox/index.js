import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './styles.css';

/**
 * Custom Checkbox component used as part of the <Checkbox> component.
 * This mimics an <input /> native component mimicking the checkbox behaviour.
 */
const CustomCheckbox = ({ checked, disabled, boxClassName, checkClassName }) => (
  <span className={classnames(boxClassName, 'checkbox', disabled ? 'disabled' : '')}>
      <i className={classnames(checkClassName, 'check')}
         style={{ visibility: checked ? 'visible' : 'hidden' }} />
    </span>
);

CustomCheckbox.propTypes = {
  /**
   * Controlled value for custom checkbox. This currently supports only
   * type `boolean`. This can be further extended to support `indeterminate`.
   * That would mean, the value of prop `checked` could then be one of types
   * [true, false, 'indeterminate'] and an `indeterminate` type could just mean
   * a different classname, this is because for the sake of simplicity of this
   * component for being the first draft version.
   */
  checked: PropTypes.bool,
  /**
   * Any additional css classes for the checkbox `box` container
   */
  boxClassName: PropTypes.string,
  /**
   * Any additional css classes for the `check` style and not the container.
   */
  checkClassName: PropTypes.string,
  /**
   * If the element needs to be disabled
   */
  disabled: PropTypes.bool,
};

CustomCheckbox.defaultProps = {
  checked: false,
};

export default CustomCheckbox;
