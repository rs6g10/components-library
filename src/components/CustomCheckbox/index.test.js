import React from 'react';
import renderer from 'react-test-renderer';

import CustomCheckbox from '.';

describe('CustomCheckbox', () => {
  it('default input to match snapshot', () => {
    const tree = renderer.create(<CustomCheckbox />);
    expect(tree).toMatchSnapshot();
  });

  it('checkbox with unchecked value to match snapshot', () => {
    const tree = renderer.create(<CustomCheckbox checked={false} name="Example" />);
    expect(tree).toMatchSnapshot();
  });
});
