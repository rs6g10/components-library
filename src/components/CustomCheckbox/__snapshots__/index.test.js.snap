// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCheckbox checkbox with unchecked value to match snapshot 1`] = `
<span
  className="checkbox"
>
  <i
    className="check"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  />
</span>
`;

exports[`CustomCheckbox default input to match snapshot 1`] = `
<span
  className="checkbox"
>
  <i
    className="check"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  />
</span>
`;
